### ISSUE TITLE
Quarto integration

### ISSUE THREAD

**Issue by agitter** [manubot/manubot#332]
Quarto describes itself as "an open-source scientific and technical publishing system built on Pandoc":
- [website](https://quarto.org/)
- [GitHub](https://github.com/quarto-dev/quarto-cli)

I've only briefly looked through its features.  It uses standard Pandoc [citations](https://quarto.org/docs/authoring/footnotes-and-citations.html), so it may be able to benefit from Manubot's citation by identifier.

**Comment by dhimmel** [comment]
Yes, I think a quarto extension to enable manubot's citation by identifier could be popular. Haven't really assessed the feasibility of this but here are some rudimentary links:

- https://quarto.org/docs/extensions/
- https://github.com/quarto-ext

It also might be interesting to see if Manubot could be built on top of quarto, where manubot would become mostly the extra tooling needed for:

- citation by identifier
- continuous integration and deployment
- git based versioning

**Comment by agitter** [comment]
A basic question (that may be answered in the docs I haven't read carefully enough) is whether extensions are pandoc filters or something quarto specific.  [Creating filters](https://quarto.org/docs/extensions/filters.html) leads me to believe they are pandoc filters.  One next step could be to take some existing quarto document and see what happens if we try to run the Manubot filter on it.

**Comment by cderv** [comment]
Hello. I am available if you have any question related to how Quarto works. 

Thanks @VladimirAlexiev for the answer. We do use Pandoc's Lua filter with Quarto internally but also to extend the feature available with some Extensions made by the community. It works with the Pandoc Lua API and also with some addition by the Quarto LUA API. (https://quarto.org/docs/extensions/lua-api.html)

> One next step could be to take some existing quarto document and see what happens if we try to run the Manubot filter on it

Did you try that ? Any filter that works with Pandoc should be able to be applied on a Quarto document. Maybe some ajustement would be needed if there are some conflicts between what you filter expect, and some processing that Quarto could have made before the filter is ran. 

Anyway, happy to help.

**Comment by agitter** [comment]
I started exploring this with Quarto 1.2.313 and the example document in https://quarto.org/docs/get-started/authoring/text-editor.html. I created a new conda environment with Jupyter, the required Python packages, and manubot (which contains the `pandoc-manubot-cite` Pandoc filter).

Quarto/Pandoc seem to recognize `pandoc-manubot-cite` when I list it as one of the `filters` in the YAML block at the beginning of the document. Listing a gibberish name for a filter instead fails with an error
```
Error running filter asdfljasdfjdl:
Could not find executable asdfljasdfjdl
```

That answers the question of whether Quarto can find the Python-based Pandoc filter. The citation-by-identifier of the `pandoc-manubot-cite` doesn't actually work yet, so that will take some debugging. I briefly experimented with running `pandoc-manubot-cite` [before or after](https://quarto.org/docs/extensions/filters.html#activating-filters) the `quarto` filter, which didn't seem to matter but should be tested further.

@dhimmel the filter is designed to work on a standalone markdown document processed with Pandoc, right? I'm now hopeful that with the right metadata it will work within Quarto.

**Comment by agitter** [comment]
I resumed testing by running the `pandoc-manubot-cite` Pandoc filter with a minimal example: [minimal.md](https://github.com/manubot/manubot/files/10528882/minimal.md)

I ran it with
```
pandoc -s minimal.md --filter=pandoc-manubot-cite --citeproc --output minimal.html
```
I still haven't gotten it to work with Quarto though: [minimal.qmd.txt](https://github.com/manubot/manubot/files/10528890/minimal.qmd.txt) (fake `.txt` extension added for GitHub upload)

I set the `manubot-output-citekeys` metadata field, so I can confirm the filter is being run. Perhaps both the `pandoc-manubot-cite` filter and Quarto are modifying the document metadata so the filter's modifications here are being overwritten?
https://github.com/manubot/manubot/blob/d4242ffa4194e4a13a68c5f6466feff559d3f9d5/manubot/pandoc/cite_filter.py#L272-L274

@cderv after the Manubot filter modifies that metadata, would something in the Quarto processing change it?

**Comment by agitter** [comment]
This [Quarto example](https://jennalandy.github.io/remodel-your-qmd/Reports_and_Themes.html#citations) shows copying a DOI into a document, extracting the reference metadata, and having a `.bib` file creating automatically. Is that a built in part of Quarto or a feature of a specific editor?

**Comment by cderv** [comment]
> This [Quarto example](https://jennalandy.github.io/remodel-your-qmd/Reports_and_Themes.html#citations) shows copying a DOI into a document, extracting the reference metadata, and having a `.bib` file creating automatically. Is that a built in part of Quarto or a feature of a specific editor?

This is a feature of visual editor:  https://quarto.org/docs/visual-editor/technical.html#citations

It is available in RStudio IDE but also in VSCODE with the [quarto extension](https://marketplace.visualstudio.com/items?itemName=quarto.quarto).

Example: in visual mode, when typing `@` you can paste a DOI and it will open this windows with downloaded info, and which bib file to write into
![image](https://user-images.githubusercontent.com/6791940/225587683-2e63e459-5b5a-48b8-9c0b-47f3fd00e44b.png)

**Comment by VladimirAlexiev** [comment]
@agitter these provide similar functionality with "plain" pandoc, without the interactivity:
- https://github.com/aeroreyna/pandoc-doi2bib
- https://github.com/pandoc/lua-filters/tree/master/doi2cite
